connect/eaas/runner/managers/transformation.py

```diff
@@ -0,0 +1,282 @@
+            result_message.output.runtime = time.monotonic() - begin_ts
+            logger.info(
+                f'Transformation task {task_data.options.task_id} '
+                f'result: {result.status}, '
+                f'took: {result_message.output.runtime}',
+            )
+            if result.status in (ResultType.SKIP, ResultType.FAIL):
+                result_message.output.message = result.output
+        except Exception as e:
+            self.log_exception(task_data, e)
+            result_message.output = TaskOutput(result=ResultType.RETRY)
+            result_message.output.message = traceback.format_exc()[:4000]
+
+        return result_message
+
+    async def process_transformation(self, task_data, tfn_request, method):
+        download_result = await asyncio.get_running_loop().run_in_executor(
+            self.executor,
+            self.download_exel,
+            tfn_request,
+        )
+        input_file = download_result.reslut()
+        output_file = NamedTemporaryFile(
+            suffix=tfn_request['files']['input']['name'].split('.')[-1],
+        )
+
+        read_queue = asyncio.Queue(TRANSFORMATION_TASK_MAX_PARALLEL_LINES)
+        write_queue = asyncio.Queue()
+
+        reader_task = asyncio.get_running_loop().run_in_executor(
+            self.executor,
+            self.read_excel,
+            input_file.name,
+            read_queue,
+        )
+        writer_task = asyncio.get_running_loop().run_in_executor(
+            self.executor,
+            self.write_excel,
+            output_file.name,
+            write_queue,
+            tfn_request['stats']['total'],
+            task_data.input.object_id,
+        )
+        processor_task = asyncio.create_task(self.process_rows(
+            read_queue, write_queue, method, total_rows,
+        ))
+
+        try:
+            await asyncio.gather([reader_task, writer_task, processor_task])
+            self.send_output_file(task_data, input_file)
+            input_file.close()
+        except Exception as e:
+            self.log_exception(task_data, e)
+            raise
+
+    def download_excel(self, tfn_request):
+        client = httpx.AsyncClient()
+        client.headers.update({
+            'API_KEY': self.config.api_key,
+            **self.config.get_user_agent(),
+        })
+
+        input_file_name = tfn_request['files']['input']['name']
+        input_file = NamedTemporaryFile(suffix=input_file_name.split('.')[-1])
+
+        with client.stream('GET', f'{self.config.get_api_url()}{input_file_name}') as response:
+            for chunk in response.iter_bytes():
+                input_file.write(chunk)
+
+        return input_file
+
+    def read_excel(self, filename, queue):
+        wb = load_workbook(filename=filename, read_only=True)
+        ws = wb.worksheets[0]
+        for idx, row in enumerate(ws.rows, start=1):
+            asyncio.run_coroutine_threadsafe(queue.put((idx, row)))
```

Comment:
Probably the second argument (loop) of `run_coroutine_threadsafe` is mandatory so it should be: 

```
asyncio.run_coroutine_threadsafe(queue.put((idx, row)), asyncio.get_running_loop())
```